internal/ethapi/transaction_options.go

```diff
@@ -0,0 +1,81 @@
+package ethapi
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/state"
+)
+
+type AccountStorage struct {
+	RootHash  *common.Hash
+	SlotValue map[common.Hash]common.Hash
+}
+
+func (r *AccountStorage) UnmarshalJSON(data []byte) error {
+	var hash common.Hash
+	if err := json.Unmarshal(data, &hash); err == nil {
+		r.RootHash = &hash
+		return nil
+	}
+	return json.Unmarshal(data, &r.SlotValue)
+}
+
+func (r AccountStorage) MarshalJSON() ([]byte, error) {
+	if r.RootHash != nil {
+		return json.Marshal(*r.RootHash)
+	}
+	return json.Marshal(r.SlotValue)
+}
+
+type TransactionOpts struct {
+	KnownAccounts  map[common.Address]AccountStorage `json:"knownAccounts"`
+	BlockNumberMin *hexutil.Uint64                   `json:"blockNumberMin,omitempty"`
+	BlockNumberMax *hexutil.Uint64                   `json:"blockNumberMax,omitempty"`
+	TimestampMin   *hexutil.Uint64                   `json:"timestampMin,omitempty"`
+	TimestampMax   *hexutil.Uint64                   `json:"timestampMax,omitempty"`
+}
+
+func (o *TransactionOpts) Check(blockNumber uint64, timeStamp uint64, statedb *state.StateDB) error {
+	if o.BlockNumberMin != nil && blockNumber < uint64(*o.BlockNumberMin) {
+		return errors.New("BlockNumberMin condition not met")
+	}
+	if o.BlockNumberMax != nil && blockNumber > uint64(*o.BlockNumberMax) {
+		return errors.New("BlockNumberMax condition not met")
+	}
+	if o.TimestampMin != nil && timeStamp < uint64(*o.TimestampMin) {
+		return errors.New("TimestampMin condition not met")
+	}
+	if o.TimestampMax != nil && timeStamp > uint64(*o.TimestampMax) {
+		return errors.New("TimestampMax condition not met")
+	}
+	if len(o.KnownAccounts) > 1000 {
```

Comment:
Is `KnownAccounts` supposed to represent total number of smart contract accounts that is supported or something else?

Reply:
"account" represent a contract which this transaction accessed the storage of.
most of them are "smart accounts", but some might be paymasters or smart-account factory contracts.
Yes, it puts an absolute maximum on the # of accounts in a handleOps bundle.
The number "1000" was chosen arbitrarily, and might change, after we do some performance analysis of implementation, to understand the impact of this size

Reply:
Thanks @drortirosh . 
With regards to this, how exactly are the `SlotValue`s determined? 
I assume `SlotValue` is a reference to the address of the storage of the account?

Reply:
The slot value is the value returned by `eth_getStorageAt(address, position)`
(we just use the term "slot" and not "position")

Reply:
Hi @drortirosh 
To make BSC work with the eth-infinitism bundler, does BSC have to enable London hardfork? 
Or can I modify the bundler to make it work with the current BSC?